### PR TITLE
Filter bugfixes

### PR DESCRIPTION
**Pull request by sarahfossheim**
**Fixed**:
- https://github.com/HTTPArchive/httparchive.org/issues/845: Sorted alphabetically now
- https://github.com/HTTPArchive/httparchive.org/issues/846: Fixes missing techs when a category is selected
- https://github.com/HTTPArchive/httparchive.org/issues/847: "ALL" re-added as an option to the technologies dropdown
- Technology dropdown breaks when selecting 'ALL' in categories dropdown: Now shows all technologies instead of an empty list

### CODE REVIEW

File: src/js/components/filters.js [hunk 1]
```diff
@@ -88,8 +88,13 @@ class Filters {
         techSelector.before(errorMsg);
       }
 
+      /* Get a list of technologies */
+      const techs = this.technologies;
+      const sortedTechs = techs.sort((a, b) => a.technology - b.technology);
```
[the remark this answers is not in the document]
Comment (sarahfossheim) [reply src/js/components/filters.js:93]: I'll just revert back for now, and suggest we add more user-friendly dropdowns as a separate action item (so deciding on a sort order and design for it doesn't block the other bugfixes in this PR). If we keep it by origins, then we should implement a component more similar to the old report, where we can display the numbers as well.

File: src/js/components/filters.js [hunk 2]
```diff
@@ -107,7 +112,8 @@ class Filters {
   updateGeo() {
     const select = document.querySelector('select#geo');
     select.innerHTML = '';
-    this.geos.forEach((geo) => {
+    const sortedGeos = this.geos.sort((a, b) => a.geo !== b.geo ? a.geo < b.geo ? -1 : 1 : 0);
```
[the remark this answers is not in the document]
Comment (sarahfossheim) [reply src/js/components/filters.js:115]: See: https://github.com/HTTPArchive/httparchive.org/pull/848#discussion_r1587516985

For geos I'm wondering if alphabetical might make it easier to find specific countries, as that's how most country selectors on the web are sorted alphabetically. If I want to find Norway, I know I still have to scroll further when I see "Finland" but am close when I see "Netherlands" or "North Korea". Maybe a good one to look for some more feedback/user tests on, or try out two different designs?

[the remark this answers is not in the document]
Comment (rviscomi) [reply src/js/components/filters.js]: I do also like how the Data Studio select list includes a search filter. More complex to reimplement but could give us the best of both.

image

File: src/js/components/filters.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (rviscomi) [reply src/js/components/filters.js]: SGTM!

**Review by rviscomi — APPROVED** [review]
LGTM!

**Pull request merged**